includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1482,30 +1482,33 @@ public function set_payment_method_title_for_order( $order, $payment_method_type
 	 * @return array Array of keyed metadata values.
 	 */
 	protected function get_metadata_from_order( $order, $payment_type ) {
+		if ( $this instanceof UPE_Split_Payment_Gateway ) {
+			$gateway_type = 'split_upe';
+		} elseif ( $this instanceof UPE_Payment_Gateway ) {
+			$gateway_type = 'upe';
+		} else {
+			$gateway_type = 'classic';
+		}
```

Comment:
Late here, but I think we would consider another type of gateway called  the Split-UPE-with-Deferred-Intent. 

Context: paJDYF-9xf-p2